lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -106,6 +113,22 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
   }
 
+  void killAfterTimeout() {
+      ::sleep(10);
+
+      if (!done) {
```

Comment:
There isn't any synchronization between `done` and the `spawnedProcesses` set, so this code is racy, but this code could just avoid tracking the `done` flag and let the signaling code run (it will be an empty set).